src/mjolnir/adminbuilder.cc

```diff
@@ -12,53 +18,21 @@
 
 #include "config.h"
 
-/* Need to know which geos version we have to work out which headers to include */
-#include <geos/version.h>
-
-#define USE_UNSTABLE_GEOS_CPP_API
-#if GEOS_VERSION_MAJOR == 3 && GEOS_VERSION_MINOR >= 8
-#include <geos/geom/CoordinateArraySequence.h>
-#else
-#include <geos/geom/CoordinateSequenceFactory.h>
-#endif
-#include <geos/geom/Geometry.h>
-#include <geos/geom/GeometryFactory.h>
-#include <geos/geom/LineString.h>
-#include <geos/geom/LinearRing.h>
-#include <geos/geom/MultiLineString.h>
-#include <geos/geom/MultiPolygon.h>
-#include <geos/geom/Point.h>
-#include <geos/geom/Polygon.h>
-#include <geos/io/WKTReader.h>
-#include <geos/io/WKTWriter.h>
-#if GEOS_VERSION_MAJOR == 3 && GEOS_VERSION_MINOR >= 9
-#include <geos/operation/linemerge/LineMerger.h>
-#else
-#include <geos/opLinemerge.h>
-#endif
-#include <geos/util/GEOSException.h>
-
-#include <boost/optional.hpp>
-#include <boost/property_tree/ptree.hpp>
-
-using namespace geos::geom;
-using namespace geos::io;
-using namespace geos::util;
-using namespace geos::operation::linemerge;
-
 // For OSM pbf reader
 using namespace valhalla::mjolnir;
 using namespace valhalla::baldr;
-
 using namespace valhalla::midgard;
+namespace bg = boost::geometry;
+
+BOOST_GEOMETRY_REGISTER_POINT_2D(PointLL, double, bg::cs::geographic<bg::degree>, first, second)
+typedef boost::geometry::model::polygon<PointLL> polygon_t;
+typedef boost::geometry::model::multi_polygon<polygon_t> mpolygon_t;
 
 namespace {
 
 struct polygondata {
-  Polygon* polygon;
-  LinearRing* ring;
+  polygon_t polygon;
   double area;
-  int iscontained;
   unsigned containedbyid;
```

Comment:
these members weren't needed anymore.. `ring` is unnecessary with boost geometry and `isocontained` is redundant since it's only 1 if `containedbyid != 0` so all checks can be made with the latter instead